src/pipecat/processors/audio/audio_buffer_processor.py

```diff
@@ -0,0 +1,83 @@
+from pipecat.frames.frames import (AudioRawFrame, BotStartedSpeakingFrame,
+                                   BotStoppedSpeakingFrame, Frame,
+                                   UserAudioFrame, UserStartedSpeakingFrame,
+                                   UserStoppedSpeakingFrame)
+from pipecat.processors.frame_processor import FrameDirection, FrameProcessor
+
+
+class AudioBufferProcessor(FrameProcessor):
+
+    def __init__(self):
+        """
+        Initialize the AudioBufferProcessor.
+
+        This constructor sets up the initial state for audio processing:
+        - audio_buffer: A bytearray to store incoming audio data.
+        - num_channels: The number of audio channels (initialized as None).
+        - sample_rate: The sample rate of the audio (initialized as None).
+        - assistant_audio: A boolean flag to indicate if assistant audio is being processed.
+        - user_audio: A boolean flag to indicate if user audio is being processed.
+
+        The num_channels and sample_rate are set to None initially and will be
+        populated when the first audio frame is processed.
+        """
+        super().__init__()
+        self._audio_buffer = bytearray()
+        self._num_channels = None
+        self._sample_rate = None
+        self._assistant_audio = False
+        self._user_audio = False
+        print(f"ctor::AudioBufferProcessor object memory address: {id(self)}")
```

Comment:
Use `loguru` logger if you need to log anything.  :pray:

Reply:
My fault, I left a debug print statement in there. Fixed.